crates/flux-middle/src/rty/mod.rs

```diff
@@ -426,12 +426,12 @@ impl GenericArg {
 }
 
 impl Clause {
-    pub fn new(kind: ClauseKind) -> Self {
-        Clause { kind }
+    pub fn new(vars: impl Into<List<BoundVariableKind>>, kind: ClauseKind) -> Self {
+        Clause { kind: Binder::new(kind, vars.into()) }
     }
 
     pub fn kind(&self) -> ClauseKind {
-        self.kind.clone()
+        self.kind.clone().skip_binder()
```

Comment:
```suggestion
        // FIXME(nilehmann) we should deal with the binder in all the places this is used instead of blindly skipping it
        self.kind.clone().skip_binder()
```